src/components/common/TokenDisplay/index.tsx

```diff
@@ -29,11 +30,13 @@ const StyledImg = styled(TokenImg)`
 `
 
 export function TokenDisplay(props: Props): JSX.Element {
-  const { erc20, network } = props
+  const { erc20, network, showAbbreviated } = props
 
   // Name and symbol are optional on ERC20 spec. Fallback to address when no name,
   // and show no symbol when that's not set
-  const tokenLabel = `${erc20.name || erc20.address}${erc20.symbol ? ` (${erc20.symbol})` : ''}`
+  const tokenLabel = showAbbreviated
+    ? `${erc20.symbol || erc20.name || abbreviateString(erc20.address, 6, 4)}`
```

Comment:
This is meant to be the full token name.
When a token has both, it would show: `SYMBOL (name)`

In this change, you are abbreviating that in the order of preference.
I like that. Would just ask to add a comment on this behaviour as this change is a bit in conflict with the comment on lines 35,36

Also, that's pretty much what `safeTokenName` does, isn't it? Can you re-use that? https://github.com/gnosis/dex-js/blob/d01b2a5a40f9e1de1986186055e809e606a1bfd4/src/utils/format.ts#L424